### PR TITLE
[T4A3][T09-A4] Cao Wei

### PR DESCRIPTION
**Pull request by LoneGunman001**

### CODE REVIEW

**Comment by LoneGunman001** [comment]
Ready for review

**Comment by LoneGunman001** [comment]
Ready for review

File: src/seedu/addressbook/data/person/Contact.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+     */
+    public static boolean isValidContact(String test, String regex) {
+        return test.matches(regex);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
+                && this.value.equals(((Contact) other).value)); // state check
```
Comment (ZawadulFarhan) [inline src/seedu/addressbook/data/person/Contact.java:44]: This works but note the case where contacts of different classes have the same value.

**Review by ZawadulFarhan — COMMENTED** [review]
This is a good attempt.

**Pull request closed without merging**